src/coq_elpi_glob_quotation.ml

```diff
@@ -87,6 +87,23 @@ let type_gen = ref 0
 
 let is_hole x = match DAst.get x with GHole _ -> true | _ -> false
 
+let universe_level_name evd ({CAst.v=id} as lid) =
+  try Evd.universe_of_name evd id
+  with Not_found ->
+    CErrors.user_err ?loc:lid.CAst.loc
+      (Pp.(str "Undeclared universe: " ++ Id.print id ++ str "."))
+
+let sort env sigma l = match l with
+| [] -> assert false
+| [GSProp, 0] -> Sorts.sprop
+| [GProp, 0] -> Sorts.prop
+| [GUniv u, 0] -> Sorts.sort_of_univ (Univ.Universe.make u)
+| [GLocalUniv l, 0] ->
+  let u = universe_level_name sigma l in
+  Sorts.sort_of_univ (Univ.Universe.make u)
+| _ ->
```

Comment:
Can you make this one more precise? It may match also because a new `GWhatEver` is introduced, while  here you really mean `_ :: _` I believe.

Reply:
No, the increment is important also. It could be a 1 (that's the only case which is really handled currently in the Coq codebase.)